dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/Utils.java

```diff
@@ -987,21 +987,27 @@ public Object getDSpaceAPIObjectFromRest(Context context, BaseObjectRest restObj
     */
     public BaseObjectRest getBaseObjectRestFromUri(Context context, String uri) throws SQLException {
         String dspaceUrl = configurationService.getProperty("dspace.server.url");
+        String dspaceSSRUrl = configurationService.getProperty("dspace.server.ssr.url", "");
 
         // Convert strings to URL objects.
         // Do this early to check that inputs are well-formed.
         URL dspaceUrlObject;
+        URL dspaceUrlSSRObject = null;
         URL requestUrlObject;
         try {
             dspaceUrlObject = new URL(dspaceUrl);
             requestUrlObject = new URL(uri);
+            if (StringUtils.isNoneBlank(dspaceSSRUrl)) {
+                dspaceUrlSSRObject = new URL(dspaceSSRUrl);
+            }
```

Comment:
What about adding `else { dspaceUrlSSRObject = dspaceUrlObject; }` and dropping all of the subsequent is-blank tests.  IOW dspaceUrlSSRObject is never null.